include/glow/Runtime/Executor/NetworkExecutionState.h

```diff
@@ -25,6 +25,30 @@
 namespace glow {
 namespace runtime {
 
+class NetworkExecutionState;
+
+class NetworkExecutionStatePool {
+public:
+  NetworkExecutionState *getNextNetworkExecutionState() {
+    std::lock_guard<std::mutex> lock(stateLock_);
+    auto nextState = availableStates_.front();
+    availableStates_.pop_front();
+    return nextState;
+  }
+
+  void addNewState(std::unique_ptr<NetworkExecutionState> state);
+
+  void returnNetworkExecutionState(NetworkExecutionState *state) {
+    std::lock_guard<std::mutex> lock(stateLock_);
+    availableStates_.push_back(state);
+  }
+
+private:
+  std::vector<std::unique_ptr<NetworkExecutionState>> states_;
+  std::deque<NetworkExecutionState *> availableStates_;
```

Comment:
It might be more efficient just to use stack pop/push.

Reply:
Is stack more efficient than deque? If they are similar I prefer deque for follow on P2P work.

Reply:
Stack is more cache friendly as you get the one that is recently checked back. But if you prefer dequeue we can check this in and measure later.